R/config.R

```diff
@@ -0,0 +1,976 @@
+# Kaiaulu - https://github.com/sailuh/kaiaulu
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+########## Configuration File Getter Functions ##########
+
+##### General Getter Functions #####
+
+#' Returns the parsed configuration file (.yml).
+#'
+#' @description The input file is expected to be in the .yml format.
+#' The function returns a parsed version of the input .yml file, and it will
+#' inform the user if the input .yml file path does not exist. The contents
+#' of the input .yml file may contain machine-dependent paths that may need to
+#' be modified by the user.
+#'
+#' @param config_path The path of the config file from the kaiaulu directory (e.g. "conf/kaiaulu.yml").
+#' @return The parsed config file whose path is specified by `config_path`.
+#' @export
+get_parsed <- function(config_path) {
+
+  conf <- yaml::read_yaml(config_path)
+
+  if (is.null(conf)) {
+    warning("Path does not exist.")
+  }
+
+  return(conf)
+}
+
+#' Returns the specified tool project from a parsed tool configuration file.
+#'
+#' @description This function returns a path to a specified tool from a
+#' specified parsed tool configuration file. The function takes the input
+#' `tool_name` and uses it to index a specific tool project in a parsed
+#' tool configuration file, `config_file`, where it then returns the specified
+#' tool project. The function will inform the user if the specified attribute,
+#' `tool_name`, exists in the parsed configuration file, `config_file`.
+#'
+#' @param tool_name The name of the tool (e.g. "perceval" or "dv8").
+#' @param config_file The parsed configuration file.
+#' @return The specified `tool_name` tool project from `config_file`.
+#' @export
+get_tool_project <- function(tool_name, config_file) {
+
+  tool_path <- config_file[[tool_name]]
+
+  if (is.null(tool_path)) {
+    warning("Attribute does not exist.")
+  }
+
+  return(tool_path)
+}
+
+##### Git Getter Functions #####
```

Comment:
Can you ensure these sections match the hierarchy shown on the new config file? For example, under tools, we would find dv8 for example. If the config specification hierarchy matches the section hierarchy, it will be much easier to find the associated get functions to that region of that config. I would even sort the function definition here to match that of the config.